parsers/GT.py

```diff
@@ -1,163 +1,206 @@
     session: Session | None = None,
     target_datetime: datetime | None = None,
     logger: Logger = getLogger(__name__),
-):
-    """Fetch a list of hourly production data, in MW, for the day of the
-    requested date-time.
-    """
+) -> list[dict]:
+    """Fetch a list of hourly production data, in MW, for the UTC day of the requested date-time."""
 
+    session = session or Session()
     target_datetime = (
-        datetime.now(TIMEZONE)
+        datetime.now(timezone.utc)
         if target_datetime is None
-        else target_datetime.astimezone(TIMEZONE)
+        else target_datetime.astimezone(timezone.utc)
     )
-
-    api_data = _get_api_data(session, URL, target_datetime, target=ApiKind.PRODUCTION)
-
-    day_datetime = datetime.combine(  # truncate to day
-        target_datetime, time(), tzinfo=TIMEZONE
+    api_data = _get_api_data(
+        session,
+        target_datetime=target_datetime,
+        target=ApiKind.PRODUCTION,
+        num_backlog_days=0,
     )
-    results = [
-        {
-            "datetime": day_datetime + timedelta(hours=hour),
-            "production": {
-                "biomass": row["BIOGAS"] + row["BIOMASA"],
-                "coal": row["CARBÓN"],
-                "gas": row["GAS NATURAL"],
-                "geothermal": row["VAPOR"],
-                "hydro": row["AGUA"],
-                "oil": row["BUNKER"] + row["DIESEL"],
-                "solar": row["IRRADIACIÓN"],
-                "unknown": row["BIOMASA/CARBÓN"]
-                + row["CARBÓN/PETCOKE"]
-                + row["SYNGAN"],
-                "wind": row["VIENTO"],
-            },
-            "source": DOMAIN,
-            "zoneKey": zone_key,
-        }
-        for hour, row in enumerate(api_data)
-    ]
-    return [
-        validation.validate(
-            result, logger, floor=PRODUCTION_THRESHOLD, remove_negative=True
-        )
-        for result in results
-    ]
-
 
-def _get_api_data(
-    session: Session | None,
-    url: str,
-    date_time: datetime,
-    target: ApiKind,
-) -> list[dict]:
-    """Get the JSON-formatted response from the AMM API for the desired date-time."""
-    session = session or Session()
-    response_payload = session.get(
-        url, params={"dt": date_time.strftime("%d/%m/%Y")}
-    ).json()
-
-    # The JSON data returned by the API is a list of objects, each
-    # representing one technology type. Collect this information into a list,
-    # with the list index representing the hour of day.
-    results = [collections.defaultdict(float) for _ in range(24)]
-    for row in response_payload:
-        # The API returns hours in the range [1, 24], so each one refers to the
-        # past hour (e.g., 1 is the time period [00:00, 01:00)). Shift the hour
-        # so each index represents the hour ahead and is in the range [0, 24),
-        # e.g., hour 0 represents the period [00:00, 01:00).
-        results[int(row["hora"]) - 1][row["tipo"]] = row["potencia"]
```

Comment:
thought: should we rather timestamp the data a bit like the API is doing e.g. timestamp the `[00:00, 01:00)` production values as being the production at `1:00` rather than the production at `00:00` ?

Reply:
We use forward looking dates internally so in this case the production for the 00:00 to 01:00 period should be marked with time of 00:00. Was it not doing that before?

Reply:
Perfect - thanks for the clarification! Yes, that is what it was doing (and still is)